tests/test_exodus_push_task.py

```diff
@@ -29,12 +32,12 @@ def test_exodus_push_typical(mock_popen, successful_gw_task, caplog):
     )
     mock_popen.return_value.wait.return_value = 0
 
+    src = os.path.join(TEST_DATA, "source-1")
     args = [
         "--debug",
         "-vvv",
         "--dry-run",
-        "/path/to/my-source,/path/to/my-dest",
-        "/path/to/other-source,/path/to/other-dest",
+        "staged:%s" % src,
```

Comment:
Will content always be in 'staged'? I don't see any other examples.

Reply:
With my confusion about pushsource aside, the question I'm getting to is: if there are other backends, should we test some besides "staged" here as well or would that just be duplicating tests in pushsource?

Reply:
The code should aim to work with any pushsource backend, even though we only expect to use it with staged & cdn-utils-staged for the initial use-cases.  With the way libraries are divided up now, the idea is that code for pushing content should be totally independent from the code for looking up the content to be pushed, such that you can mix and match them in whatever way is needed.  It's reasonable for tests to generally use 'staged' source though because that's an easy way to set up realistic inputs, given that you can just create a certain directory structure and don't need to involve any remote systems.

A related thought I had while looking at test_exodus_push_unexpected_push_item, though ultimately decided it wasn't important enough to mention.  That test doesn't need to be mocking Source.get, since pushsource library is designed in a way which allows you to register your own backend.  You can even do it in a single line returning a list of arbitrary push items, like:

```python
    Source.register_backend("test", lambda: [PushItem(name="quux")])
```

...and you could then invoke the entry point with 'test:' as the source and see how it behaves.  So you really don't need to mock.

IMO changing it not to mock isn't strictly needed, but I mention this since if there was at least one test working like that, it would make it clearer that the code should work with arbitrary pushsource backends and not be specific to 'staged'.